internal/crypto/ptls/secure_profile_min_tls_version_for_nonfips_12.go

```diff
@@ -0,0 +1,10 @@
+// Copyright 2024 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build secure_profile_min_tls_for_nonfips_12
+
+package ptls
+
+import "crypto/tls"
+
+const SecureProfileMinTLSVersionForNonFIPS = tls.VersionTLS12
```

Comment:
this name does not match the value defined in internal/crypto/ptls/profiles_fips_strict.go L76,
config.MaxVersion = DefaultProfileMaxTLSVersionForFIPS

Reply:
Hi @luwangVMW,

There are two new build tags:

1. `secure_profile_min_tls_for_nonfips_12` is only for non-FIPS builds. By default, there is a global const called `SecureProfileMinTLSVersionForNonFIPS` set to 1.3. When the build tag is used during compilation, it changes the value of that const to be 1.2. This variable is only used in `internal/crypto/ptls/profiles.go` which is the file used in non-FIPS builds.
2. `default_profile_max_tls_for_fips_13` is only for FIPS builds. By default, there is a global const called `DefaultProfileMaxTLSVersionForFIPS` set to 1.2. When the build tag is used during compilation, it changes the value of that const to be 1.3. This variable is only used in `internal/crypto/ptls/profiles_fips_strict.go` which is the file used in FIPS builds.

Side note: In the future, I expect that the official golang compiler will use a version of boringcrypto which supports TLS 1.3. The golang maintainers previously estimated that this could happen in Go 1.24 in February 2025. If that happens, then we may be able to enable TLS 1.3 by default when compiled in FIPS mode, so we may not need the setting `default_profile_max_tls_for_fips_13` described above at all because it would be the same as our new default.